Automattic/jetpack
# Connection: Update ConnectionErrorNotice React component to use message property

#### Changes proposed in this Pull Request:
Update the `ConnectionErrorNotice` React component to support and require a message property. The logic to generate the message is removed from the component and moved to the places where the component is utilized. This is part of a bigger project (1202697549597858-as) to provide consistent failed connection error reporting and an interface to reconnect (1202697549597858-as-1202697549597864). 

The changes will affect the following:
* Connection

#### Other information:


#### Jetpack product discussion
* p9dueE-5w8-p2

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:
* Modify `projects/packages/my-jetpack/_inc/components/my-jetpack-screen/index.jsx` to have the below changes:
```
diff --git a/projects/packages/my-jetpack/_inc/components/my-jetpack-screen/index.jsx b/projects/packages/my-jetpack/_inc/components/my-jetpack-screen/index.jsx
index 6864b0d54d..a08566223a 100644
--- a/projects/packages/my-jetpack/_inc/components/my-jetpack-screen/index.jsx
+++ b/projects/packages/my-jetpack/_inc/components/my-jetpack-screen/index.jsx
@@ -9,6 +9,7 @@ import {
 import { Notice } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { Icon, warning, info } from '@wordpress/icons';
+import { ConnectionErrorNotice, useConnection } from '@automattic/jetpack-connection';
 import React, { useEffect } from 'react';
 import useAnalytics from '../../hooks/use-analytics';
 import useConnectionWatcher from '../../hooks/use-connection-watcher';
@@ -52,6 +53,10 @@ export default function MyJetpackScreen() {
                recordEvent( 'jetpack_myjetpack_page_view' );
        }, [ recordEvent ] );
 
+       const { connectionErrors } = useConnection( {} );
+       const connectionErrorList = Object.values( connectionErrors ).shift();
+       const connectionErrorMessage = connectionErrorList && connectionErrorList.length && connectionErrorList[ 0 ].error_message;
+
        // No render when site is not connected.
        const { isSiteConnected } = useMyJetpackConnection();
 
@@ -73,6 +78,11 @@ export default function MyJetpackScreen() {
                                                        { __( 'Manage your Jetpack products', 'jetpack-my-jetpack' ) }
                                                </Text>
                                        </Col>
+                                       { Boolean( connectionErrorMessage ) && (
+                                               <Col>
+                                                       <ConnectionErrorNotice message={ connectionErrorMessage } />
+                                               </Col>
+                                       ) }
                                        { message && (
                                                <Col>
                                                        <GlobalNotice message={ message } options={ options } clean={ clean } />
```
* Run `jetpack build packages/my-jetpack`.
* Enable the "Jetpack Debug Tools" plugin.
* Go to Jetpack Debug > Jetpack Debug.
* Enable "Broken token Utilities" and click "Save".
* Go to Jetpack Debug > Connection Errors
* Under "Create fake errors", select "With the blog token", select "Verified", and click the "Create error" button.
* Go to Jetpack > My Jetpack.
* You should see an error message as shown in the below screenshot:
![jetpack-connection-error-example](https://user-images.githubusercontent.com/56307/190708871-b9c573e3-0e90-451c-8fc0-7de2ebcd8d2e.png)
